src/animations/up-toggle.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { motion } from 'framer-motion';
+import { useState } from 'react';
+import { useEffect } from 'react';
+
+interface UpToggleProps {
+  className?: string;
+  showDefault: boolean;
+  defaultContent: React.ReactNode;
+  alternateContent: React.ReactNode;
+}
+
+export const UpToggle = ({
+  className = '',
+  showDefault = true,
+  defaultContent,
+  alternateContent,
+}: UpToggleProps) => {
+  // prevents transition on mount
+  const [inactiveY, setInactiveY] = useState(0);
+  const [inactiveOpacity, setInactiveOpacity] = useState(1);
+
+  // enables transition after mount
```

Comment:
for future reference, this can get messy if we have a lot of initial states we want to set

if it becomes too complex, i think a single `isFirstRender` state can simplify this since if `initial={false}` it goes to takes whatever is in `animate` to be default

Reply:
That is a way cleaner solution. I changed it to that. Thanks for the callout here.